indexclient/client.py

```diff
@@ -372,6 +372,9 @@ def __ne__(self, other_doc):
         """
         return self._sorted_doc != other_doc._sorted_doc
 
+    def __hash__(self):
+        return hash(repr(self))
```

Comment:
I think you should limit this to properties that are fixed, like the `did` and `baseid`. So even if the document get updated, it does not invalidate the `hash`. Something like 
``` python
def __hash__(self):
    return hash((self.did, self.baseid))
```

Reply:
I synced up with @kulgan offline. We agreed to use `did` as an entity that _should not_ change on the Document, so it shouldn't have weird behavior. 

Note this means that hypothetically if you did a request to retrieve a document with `did=123`, added it to a set, updated the document in some other attribute, then refetched the document, tried to add it to the same set, you'd have the same hash and thus have the old version of the document.

Reply:
Also note `__hash__` is going to be different than `__eq__`, which currently checks _all_ properties.

Reply:
I don't recall anywhere we check for document equality but its possible we did somewhere. I believe the intention of this equality impl was meant for special need or some generic intuition and can be refactored into something like `props_equal`. The `did` alone should be sufficient to uniquely identify a document even if other properties changed, mixing it with the `baseid` can make it more reliable. Property changes just represents a different state of the same object, and not a different object.

Reply:
Using `did` as the document identity for `__eq__`/`__hash__` isn't a bad idea (assuming we make `did` a read-only attribute), but it'd be a breaking change to indexclient, so my inclination would be to first see how much effort it'd be to fix the API behavior instead.